services/skus/controllers.go

```diff
@@ -623,15 +623,19 @@ type createItemCredsRequest struct {
 // createItemCreds handles requests for creating credentials for an item.
 func createItemCreds(svc *Service) handlers.AppHandler {
 	return func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
-		ctx := r.Context()
-		lg := logging.Logger(ctx, "skus.createItemCreds")
+		b, err := io.ReadAll(io.LimitReader(r.Body, reqBodyLimit10MB))
+		if err != nil {
+			return handlers.WrapError(err, "error reading body", http.StatusBadRequest)
+		}
 
-		req := &createItemCredsRequest{}
-		if err := requestutils.ReadJSON(ctx, r.Body, req); err != nil {
-			lg.Error().Err(err).Msg("failed to read body payload")
-			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		var req createItemCredsRequest
```

Comment:
`req` was a pointer before this change, and now it's a value. It would have been alright, had it not been passed around as a whole, which it is (to the validator). Prior to the change it was passed by pointer hence no copy, with the change it becomes a copy.

It is not a direct issue now, because the only member of the struct is a slice, but could be later on, if the data structure does change.

What do you think of changing it to:

```suggestion
		req := &createItemCredsRequest{}
```

Or explicitly passing to `ValidateStruct` by address?